### ISSUE TITLE
Enabling Tcl suppresses errors when loading productions

### ISSUE THREAD

**Issue by adeschamps** [SoarGroup/Soar#274]
When I enable Tcl, the `LoadProductions` call silently fails. Consider the following example, where `load.soar` either doesn't exist or contains some sort of error:

```c++
#include <sml_Client.h>

int main()
{
    auto* kernel = sml::Kernel::CreateKernelInNewThread();
    auto* agent = kernel->CreateAgent("test-agent");

#if 0
    auto* result = kernel->ExecuteCommandLine("tcl on", agent->GetAgentName());
    std::cerr << "\"tcl on\" result: " << result << "\n";
#endif
    
    bool loaded = agent->LoadProductions("load.soar");
    if (!loaded || agent->HadError()) {
        std::cerr << "Error: " << agent->GetLastErrorDescription() << "\n";
    } else {
        std::cerr << "No errors ???\n";
    }
}
```

If I enable tcl and load productions through either the CLI or Java debugger, I see the errors I would expect. In my program, I've tried registering for various events (`smlEVENT_ECHO`, `smlEVENT_PRINT`) but with no luck.

**Comment by amininger** [comment]
Hi Anthony,
As a workaround, try using the command:
agent->ExecuteCommandLine("source load.soar")
instead of LoadProductions. This will return a string which should have the
information you need.
I'll look into this, my guess is that the TCL interpreter doesn't hook into
the error descriptions when it fails.
-Aaron Mininger


On Mon, Feb 27, 2017 at 2:40 PM, Anthony Deschamps <notifications@github.com

**Comment by adeschamps** [comment]
Thanks Aaron, that works well enough for me. I may not be able to halt the program, but at least I see the output in the console.

**Comment by mazina** [comment]
@amininger fixed in commit 1d55002969ca38aa901cdc9c1d492ba231493f9a
